doc/tutorials/using_partition_mode_on_up2.rst

```diff
@@ -2,154 +2,311 @@
+           .pdev = {
+              .bdf.bits = {.b = 0x00U, .d = 0x12U, .f = 0x0U},
+              .bar = {
+              [0] = {
+                .base = 0x91514000UL,
+                .size = 0x2000UL,
+                .type = PCIBAR_MEM32
+              },
+              [1] = {
+                .base = 0x91537000UL,
+                .size = 0x100UL,
+                .type = PCIBAR_MEM32
+              },
+              [5] = {
+                .base = 0x91536000UL,
+                .size = 0x800UL,
+                .type = PCIBAR_MEM32
+              },
+           }
+           ...
+           static struct vpci_vdev_array vpci_vdev_array2 = {
+              .num_pci_vdev = 3,
+              ...
+              {/*vdev 1: USB controller*/
+                 .vbdf.bits = {.b = 0x00U, .d = 0x01U, .f = 0x0U},
+                 .ops = &pci_ops_vdev_pt,
+                 .bar = {
+                    [0] = {
+                      .base = 0UL,
+                      .size = 0x10000UL,
+                      .type = PCIBAR_MEM32
+                    },
                  },
-              }
-       };
-
-* Configure ``.bootargs``
-
-  A default command line is pre-configured in the source code. Here we just
-  configure ``root=`` option to help kernel locate the right rootfs and keep the
-  other options as default.
+                 .pdev = {
+                    .bdf.bits = {.b = 0x00U, .d = 0x15U, .f = 0x0U},
+                    .bar = {
+                      [0] = {
+                        .base = 0x91500000UL,
+                        .size = 0x10000UL,
+                        .type = PCIBAR_MEM64
+                      },
+                    }
+              ...
 
-  .. code-block:: none
+#. Optional configure the ``.bootargs`` kernel command line arguments
 
-     @@ -171,7 +180,8 @@ struct vm_description_array vm_desc_partition = {
-                                     .start_hpa = 0x100000000UL,
-                                     .mem_size = 0x20000000UL, /* uses contiguous memory from host */
-                                     .vm_vuart = true,
-     -                               .bootargs = "root=/dev/sda rw rootwait noxsave maxcpus=2 nohpet console=hvc0 \
-     +                               .bootargs = "root=PARTUUID=<Your rootfs PARTUUID> rw rootwait noxsave maxcpus=2 nohpet console=hvc0 \
-                                                   console=ttyS0 no_timer_check ignore_loglevel log_buf_len=16M \
-                                                   consoleblank=0 tsc=reliable xapic_phys",
-                                     .vpci_vdev_array = &vpci_vdev_array1,
+   The kernel command line arguments used to boot the privileged VMs are
+   also hardcoded for now. In case you plan to use your customized root
+   filesystem, you may optional edit the ``root=`` parameter specified
+   in the ``.bootargs`` field of the ``.vm_desc_array`` structure, to
+   instruct the Linux kernel mounts the right disk partition:
```

Comment:
I'm assuming that using `/dev/sda3` is correct for both VMs because they each only see one disk which is then recognized as `/dev/sda`, is that correct? Maybe you can add a small note to make this more explicit?

Reply:
You are right, no need to change the ``.bootargs`` if both SATA and USB disks installed with Clear Linux automatic installation.